Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -257,14 +246,21 @@ + (NSData *)generateLocalProfileKey
 
 #pragma mark - Local Profile
 
-- (NSString *)localProfileName
+- (NSData *)localProfileKey
+{
+    OWSAssert(self.localUserProfile.profileKey.length == kProfileKeyLength);
+
+    return self.localUserProfile.profileKey;
+}
+
+- (nullable NSString *)localProfileName
 {
     OWSAssert([NSThread isMainThread]);
 
-    return [self getOrCreateLocalUserProfile].profileName;
+    return self.localUserProfile.profileName;
 }
 
-- (UIImage *)localProfileAvatarImage
+- (nullable UIImage *)localProfileAvatarImage
 {
     OWSAssert([NSThread isMainThread]);
 
```

Comment:
I think there's going to be lots of small bugs like this, since I haven't been able to test yet.  Moxie suggested testing by modifying the encrypt/decrypt methods to return inputs for now.  I'm going to make that change today - hopefully it'll help flush out the worst issues.